src/actions/AddSiteActions.js

```diff
@@ -72,3 +91,42 @@ export const resetAddScreenFields = () => {
         type: ADD_SITE_FIELDS_RESET
     }
 };
+
+export const checkIfSiteIsReadyForUpload = () => {
+    return {
+        type: CHECK_IF_SITE_IS_READY
+    }
+};
+
+
+export const attemptToUploadSite = ({title, description, directions, nearestTown, accessibility, facilities, coordinate}) => {
+    const {longitude, latitude} = coordinate;
+    const uniqueTitle = `${title}${longitude}${latitude}`;
+
+    return (dispatch) => {
+        firebase.firestore().doc(`campsites/${uniqueTitle}`)
+            .set({
+                title,
+                description,
+                directions,
+                nearestTown,
+                accessibility,
+                facilities,
+                coordinate
+            })
+            .then(() => {
+                dispatch({
+                    type: ADD_SITE_SUCCESS
+                });
+            })
+            .catch(error => {
+                dispatch({
+                    type: ADD_SITE_FAILURE,
+                    payload: {error}
+                });
+            });
```

Comment:
Grab everything that was sent over for the upload... try to upload it, and send off the appropriate message to the reducers depending on what happened.